S21/indraneelL/refactorme.py

```diff
@@ -1,84 +1,92 @@
 import unicodedata
 
+class BaseClass:
+    CAPITAL_ACUTE_A = '\u00c1'
+    CAPITAL_ACUTE_E = '\u00c9'
+    CAPITAL_ACUTE_I = '\u00cd'
+    CAPITAL_ACUTE_O = '\u00d3'
+    CAPITAL_ACUTE_U = '\u00da'
+    CAPITAL_I = '\u0049'
+    DOTLESS_i = '\u0131'
+    CAPITAL_SIGMA = '\u03a3'
+    SMALL_SIGMA = '\u03c2'
+
+    STANDARD_VOWELS = 'AEIOU' + CAPITAL_ACUTE_A + CAPITAL_ACUTE_E + CAPITAL_ACUTE_I + CAPITAL_ACUTE_O + CAPITAL_ACUTE_U
+
+    def __init__(self,word):
+        self.word = word
+
+class CheckIrishWord(BaseClass):
+
+    def process_word(self):
+        _word = self.word
+        if len(self.word)>1:
+            if (self.word[0]=='t' or self.word[0]=='n') and unicodedata.normalize('NFC', self.word)[1] in self.STANDARD_VOWELS:
+                _word = self.word[0]+'-'+_word[1:]
+        return _word.lower()
+
+    def isLenited(self):
+        if len(self.word) < 2:
+            return False
+        else:
+            return self.word[0].lower() in 'bcdfgmpst' and self.word[1].lower()=='h'
+
+class CheckTurkAndAzerWord(BaseClass):
+
+    def process_word(self):
+        _word = self.word
+        _word = _word.replace(self.CAPITAL_I,self.DOTLESS_i)
+        return _word.lower()
+
+class CheckGreekWord(BaseClass):
+
+    def process_word(self):
+        _word = self.word
+        if _word[-1]== self.CAPITAL_SIGMA:
+            _word = _word[:-1]+ self.CAPITAL_SIGMA
+        return _word.lower()
+
 class Word:
```

Comment:
Didn't expect to see this since you already have BaseClass.  What's the connection between Word and BaseClass?  Combine them?

Reply:
oh I think I understand better now.   You might consider renaming BaseClass to something like "LowercaseStrategy" since you're lumping together the algorithms, independent of language, in its subclasses.  (Also move the aliases for special characters into subclasses).